pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -165,12 +167,16 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 			logger.Errorf("Failed to update TaskRun status for PipelineRun %s: %v", pr.Name, err)
 			return c.finishReconcileUpdateEmitEvents(ctx, pr, before, err)
 		}
-		go func(metrics *Recorder) {
-			err := metrics.DurationAndCount(pr)
-			if err != nil {
-				logger.Warnf("Failed to log the metrics : %v", err)
-			}
-		}(c.metrics)
+
+		if equality.Semantic.DeepEqual(original.Status, pr.Status) {
```

Comment:
I don't think this is enough to prevent false metrics from being reported.
Not knowing much about opencensus, my expectation reading this code is that `DurationAndCount` was idempotent, and it would take care of not submitting twice.
Now, if the controller is restarted for instance, all metrics will be submitted again.

A way to solve this could perhaps be to write metrics where the completion time is set?